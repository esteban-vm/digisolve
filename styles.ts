import { css } from '@emotion/react'

export const globalStyles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --color-primary: #ed07b7;
    --color-dark-primary: #bd0592;
    --color-secondary: #6920b0;
    --color-tertiary: #acdaff;

    --color-white: #fff;
    --color-grey: #555;
    --color-black: #000;

    --color-light: #f4f4f4;
    --color-lighter: #fcfcfc;

    --color-overlay1: #0000004d;
    --color-overlay2: #000000b3;
    --color-overlay3: #00000080;
    --color-overlay4: #000000cc;

    --color-shadow: #efefefef;
  }

  html,
  body {
    background-color: var(--color-white);
    color: var(--color-black);
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }

  .clear_fix {
    zoom: 1;

    ::after {
      content: '';
      display: table;
      clear: both;
      visibility: hidden;
    }
  }
`

export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
