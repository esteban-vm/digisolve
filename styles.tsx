import { Global, css } from '@emotion/react'

const styles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --primary-color: #ed07b7;
    --primary-color-dark: #bd0592;
    --secondary-color: #6920b0;
    --light-color: #acdaff;
    --dark-color: #555;
  }

  html,
  body {
    background-color: #fff;
    color: #000;
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }
`

export const Globals = () => <Global styles={styles} />
export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
