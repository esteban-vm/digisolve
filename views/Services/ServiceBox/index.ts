export { default, type ServiceProps } from './ServiceBox'
