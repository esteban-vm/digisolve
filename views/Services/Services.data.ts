import type { ServiceProps } from './Service'

const description = 'Ipsam reiciendis eligendi est qui itaque.'

const services: ServiceProps[] = [
  {
    id: '99a36a03-7210-44c9-b031-f0e2acec54e5',
    title: 'branding',
    icon: 'tags',
    description,
  },
  {
    id: '82879f1c-49d5-4b10-b419-043556c4aa3b',
    title: 'content',
    icon: 'pen-clip',
    description,
  },
  {
    id: '7ec3da85-2944-4097-b198-f1fc7a0863a7',
    title: 'marketing',
    icon: 'filter-circle-dollar',
    description,
  },
  {
    id: 'fba92296-7d8b-4150-b10d-2f1a90041c65',
    title: 'web design',
    icon: 'code',
    description,
  },
]

export default services
