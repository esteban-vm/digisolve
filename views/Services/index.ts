export { default as Services, services } from './Services'
