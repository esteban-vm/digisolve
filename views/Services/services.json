[
  { "title": "branding", "icon": "tags" },
  { "title": "content", "icon": "pen-alt" },
  { "title": "marketing", "icon": "funnel-dollar" },
  { "title": "web design", "icon": "laptop-code" }
]
