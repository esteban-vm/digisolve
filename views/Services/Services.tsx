import type { FunctionalComponent } from '@/types'
import { ServiceItem, type ServiceProps } from '@/components'
import { faTags, faPenAlt, faFunnelDollar, faLaptopCode } from '@fortawesome/free-solid-svg-icons'
import { styled } from '@/utils/styles'

export const services: ServiceProps[] = [
  { heading: 'Branding', icon: faTags },
  { heading: 'Content', icon: faPenAlt },
  { heading: 'Marketing', icon: faFunnelDollar },
  { heading: 'Web Design', icon: faLaptopCode },
]

const ServiceComponent: FunctionalComponent = (props) => {
  return (
    <section {...props}>
      <div className='row'>
        <h2 className='section_heading'>Our Services</h2>
        <h3 className='section_subheading'>Lorem ipsum dolor sit amet consectetur adipisicing elit. Omnis, ducimus.</h3>
      </div>
      <div className='row'>
        {services.map((service, index) => (
          <ServiceItem key={index} {...service} role='listitem' />
        ))}
      </div>
    </section>
  )
}

const Services = styled(ServiceComponent)`
  text-align: center;
  background-color: var(--color-light);
`

export default Services
