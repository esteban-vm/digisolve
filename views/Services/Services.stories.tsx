import type { Meta } from '@/stories'
import Services from './Services'

export default {
  component: Services,
  title: 'Views/Services',
  parameters: {
    backgrounds: {
      disable: true,
    },
  },
} as Meta<typeof Services>

export const View = () => <Services />
