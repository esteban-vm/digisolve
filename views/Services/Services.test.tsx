import { render, cleanup, screen, create, axe } from '@/tests'
import { mediaQuery } from '@/styles'
import services from './Services.data'
import Services from './Services'

describe('ðŸ§ª SERVICES:', () => {
  describe('render tests:', () => {
    let container: HTMLElement

    beforeEach(() => {
      void ({ container } = render(<Services />))
    })

    afterEach(cleanup)

    it('should be accessible', async () => {
      const results = await axe(container)
      expect(results).toHaveNoViolations()
    })

    describe('should be rendered:', () => {
      it('the title', () => {
        const title = screen.getByRole('heading', { name: /^our services$/i, level: 2 })
        expect(title).toBeInTheDocument()
        expect(title).toBeVisible()
      })

      it('the subtitle', () => {
        const subtitle = screen.getByRole('heading', { level: 3 })
        expect(subtitle).toBeInTheDocument()
        expect(subtitle).toBeVisible()
      })

      it('the services', () => {
        const serviceElements = screen.getAllByRole('article')
        expect(serviceElements).toHaveLength(services.length)

        for (const service of serviceElements) {
          expect(service).toBeInTheDocument()
          expect(service).toBeVisible()
        }
      })
    })
  })

  describe('style tests:', () => {
    let tree: ReturnType<typeof create>

    beforeEach(() => {
      tree = create(<Services />)
    })

    it('should render with correct styles', () => {
      expect(tree).toMatchSnapshot()
    })

    it('should have correct styles on extra small screen devices', () => {
      const media = mediaQuery('xs')
      const property = 'margin-bottom'
      const value = '20px'
      expect(tree).toHaveStyleRule(property, value, { target: '.xs_half:first-of-type', media })
      expect(tree).toHaveStyleRule(property, value, { target: '.xs_half:nth-of-type(2)', media })
    })
  })
})
