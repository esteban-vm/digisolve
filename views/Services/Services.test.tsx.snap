// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª SERVICES: style tests: should render with correct styles 1`] = `
.emotion-0 {
  text-align: center;
  background-color: var(--color-light);
}

@media only screen and (max-width: 480px) {
  .emotion-0 .xs_half:first-of-type,
  .emotion-0 .xs_half:nth-of-type(2) {
    margin-bottom: 20px;
  }
}

.emotion-1 {
  padding: 1% 1.5%;
}

.emotion-1 h4 {
  font-size: 150%;
  margin-bottom: 30px;
  font-weight: 400;
  text-transform: capitalize;
}

.emotion-1 p {
  color: var(--color-grey);
  font-size: 90%;
  margin-bottom: 20px;
}

.emotion-1 a:link,
.emotion-1 a:visited {
  color: var(--color-primary);
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1 a:hover {
  opacity: 0.7;
}

.emotion-1 .icon {
  margin-bottom: 30px;
  color: var(--color-secondary);
}

<section
  aria-labelledby=":r0:"
  className="emotion-0"
  id="services"
>
  <div
    className="row "
    role="presentation"
  >
    <h2
      className="section__heading"
      id=":r0:"
    >
      Our services
    </h2>
    <h3
      className="section__subheading"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Omnis, ducimus.
    </h3>
  </div>
  <div
    className="row "
    role="presentation"
  >
    <div
      className="col span_1_of_4 xs_half"
      role="presentation"
    >
      <article
        aria-labelledby="99a36a03-7210-44c9-b031-f0e2acec54e5"
        className="animate__animated animate__fast emotion-1"
      >
        <svg
          aria-hidden={true}
          className="icon"
          color="var(--color-primary)"
          fill="currentColor"
          height="3rem"
          role="img"
          viewBox="0 0 512 512"
          width="3rem"
        >
          <path
            d="M345 39.1l127.8 129.3c52.4 53 52.4 138.2 0 191.2l-112 113.3c-9.3 9.4-24.5 9.5-33.9.2s-9.5-24.5-.2-33.9l111.9-113.3c33.9-34.3 33.9-89.4 0-123.7L310.9 72.9c-9.3-9.4-9.2-24.6.2-33.9s24.6-9.2 33.9.2zM0 229.5V80c0-26.5 21.5-48 48-48h149.5c17 0 33.3 6.7 45.3 18.7l168 168c25 25 25 65.5 0 90.5L277.3 442.7c-25 25-65.5 25-90.5 0l-168-168C6.7 262.7 0 246.5 0 229.5zM144 144c0-17.7-14.3-32-32-32s-32 14.3-32 32 14.3 32 32 32 32-14.3 32-32z"
          />
        </svg>
        <h4
          id="99a36a03-7210-44c9-b031-f0e2acec54e5"
        >
          branding
        </h4>
        <p>
          Ipsam reiciendis eligendi est qui itaque.
        </p>
        <a
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Read more
        </a>
      </article>
    </div>
    <div
      className="col span_1_of_4 xs_half"
      role="presentation"
    >
      <article
        aria-labelledby="82879f1c-49d5-4b10-b419-043556c4aa3b"
        className="animate__animated animate__fast emotion-1"
      >
        <svg
          aria-hidden={true}
          className="icon"
          color="var(--color-primary)"
          fill="currentColor"
          height="3rem"
          role="img"
          viewBox="0 0 512 512"
          width="3rem"
        >
          <path
            d="M453.3 19.3l39.4 39.4c25 25 25 65.5 0 90.5l-52.1 52.1-1-1-16-16-96-96-17-17 52.1-52.1c25-25 65.5-25 90.5 0zM241 114.9c-9.4-9.4-24.6-9.4-33.9 0L105 217c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9L173.1 81c28.1-28.1 73.7-28.1 101.8 0L288 94.1l17 17 96 96 16 16 1 1-17 17-171.5 171.4c-48 48-109.2 80.8-175.8 94.1l-25 5c-7.9 1.6-16-.9-21.7-6.6s-8.1-13.8-6.6-21.7l5-25c13.3-66.6 46.1-127.8 94.1-175.8L254.1 128 241 114.9z"
          />
        </svg>
        <h4
          id="82879f1c-49d5-4b10-b419-043556c4aa3b"
        >
          content
        </h4>
        <p>
          Ipsam reiciendis eligendi est qui itaque.
        </p>
        <a
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Read more
        </a>
      </article>
    </div>
    <div
      className="col span_1_of_4 xs_half"
      role="presentation"
    >
      <article
        aria-labelledby="7ec3da85-2944-4097-b198-f1fc7a0863a7"
        className="animate__animated animate__fast emotion-1"
      >
        <svg
          aria-hidden={true}
          className="icon"
          color="var(--color-primary)"
          fill="currentColor"
          height="3rem"
          role="img"
          viewBox="0 0 576 512"
          width="3rem"
        >
          <path
            d="M3.9 22.9C10.5 8.9 24.5 0 40 0h432c15.5 0 29.5 8.9 36.1 22.9s4.6 30.5-5.2 42.5L396.4 195.6C316.2 212.1 256 283 256 368c0 27.4 6.3 53.4 17.5 76.5-1.6-.8-3.2-1.8-4.7-2.9l-64-48c-8.1-6-12.8-15.5-12.8-25.6v-79.1L9 65.3C-.7 53.4-2.8 36.8 3.9 22.9zM576 368c0 79.5-64.5 144-144 144s-144-64.5-144-144 64.5-144 144-144 144 64.5 144 144zm-167.2-32.6c.6-.9 1.8-2.1 4.2-3.4 5.1-2.7 12.5-4.1 18.7-4 8.2.1 17.1 1.8 26.4 4.1 8.6 2.1 17.3-3.1 19.4-11.7s-3.1-17.3-11.7-19.4c-5.6-1.4-11.6-2.7-17.9-3.7V288c0-8.8-7.2-16-16-16s-16 7.2-16 16v9.5c-6.1 1.2-12.3 3.2-18 6.3-11.8 6.3-23 18.4-21.8 37.2 1 16 11.7 25.3 21.6 30.7 8.8 4.7 19.7 7.8 28.6 10.3l1.8.5c10.3 2.9 17.9 5.2 23.2 8.3 4.5 2.7 4.7 4.2 4.7 5.6.1 2.4-.5 3.7-1 4.5-.6 1-1.8 2.2-4 3.3-4.7 2.5-11.8 3.8-18.5 3.6-9.5-.3-18.5-3.1-29.9-6.8-1.9-.6-3.8-1.2-5.8-1.8-8.4-2.6-17.4 2.1-20 10.5s2.1 17.4 10.5 20c1.6.5 3.3 1 5 1.6 7 2.3 15.1 4.8 23.7 6.6v11.4c0 8.8 7.2 16 16 16s16-7.2 16-16v-10.6c6.2-1.1 12.5-3.1 18.3-6.2 12.1-6.5 22.3-18.7 21.7-36.9-.5-16.2-10.3-26.3-20.5-32.3-9.4-5.6-21.2-8.9-30.5-11.5h-.2c-10.4-2.9-18.3-5.2-23.9-8.2-4.8-2.6-4.8-4-4.8-4.5v-.1c-.1-1.9.3-2.9.8-3.6z"
          />
        </svg>
        <h4
          id="7ec3da85-2944-4097-b198-f1fc7a0863a7"
        >
          marketing
        </h4>
        <p>
          Ipsam reiciendis eligendi est qui itaque.
        </p>
        <a
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Read more
        </a>
      </article>
    </div>
    <div
      className="col span_1_of_4 xs_half"
      role="presentation"
    >
      <article
        aria-labelledby="fba92296-7d8b-4150-b10d-2f1a90041c65"
        className="animate__animated animate__fast emotion-1"
      >
        <svg
          aria-hidden={true}
          className="icon"
          color="var(--color-primary)"
          fill="currentColor"
          height="3rem"
          role="img"
          viewBox="0 0 640 512"
          width="3rem"
        >
          <path
            d="M392.8 1.2c-17-4.9-34.7 5-39.6 22l-128 448c-4.9 17 5 34.7 22 39.6s34.7-5 39.6-22l128-448c4.9-17-5-34.7-22-39.6zm80.6 120.1c-12.5 12.5-12.5 32.8 0 45.3l89.3 89.4-89.4 89.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l112-112c12.5-12.5 12.5-32.8 0-45.3l-112-112c-12.5-12.5-32.8-12.5-45.3 0zm-306.7 0c-12.5-12.5-32.8-12.5-45.3 0l-112 112c-12.5 12.5-12.5 32.8 0 45.3l112 112c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256l89.4-89.4c12.5-12.5 12.5-32.8 0-45.3z"
          />
        </svg>
        <h4
          id="fba92296-7d8b-4150-b10d-2f1a90041c65"
        >
          web design
        </h4>
        <p>
          Ipsam reiciendis eligendi est qui itaque.
        </p>
        <a
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Read more
        </a>
      </article>
    </div>
  </div>
</section>
`;
